sklearn/utils/multiclass.py

```diff
@@ -221,6 +224,13 @@ def check_classification_targets(y):
             "regression target with continuous values."
         )
 
+    if n_samples is not None:
+        if len(np.unique(y)) > round(0.5 * n_samples):
```

Comment:
This runs counter to the second point in https://github.com/scikit-learn/scikit-learn/issues/16399:

> check_classification_targets which calls type_of_target triggers a call to _assert_all_finite and np.unique(y) which are redundant with checks done elsewhere 

In general, including another `np.unique(y)` adds another `n*log(n)` operation. (`np.unique` sorts the data)